src/Adapters/EventSource.Netstandard13/EventSourceTelemetryModule.cs

```diff
@@ -218,5 +229,65 @@ private void EnableAsNecessary(EventSource eventSource)
                 }
             }
         }
+
+        /// <summary>
+        /// Returns true when eventSource satisfied the rule; false otherwise. Returns false when either is null.
+        /// </summary>
+        /// <param name="eventSource">The target event source.</param>
+        /// <param name="rule">The naming rule to be used for matching.</param>
+        private bool IsEventSourceNameMatch(EventSource eventSource, EventSourceNamingMatchRuleBase rule)
+        {
+            if (rule == null || eventSource == null)
+            {
+                return false;
+            }
+
+            if (!rule.PrefixMatch)
+            {
+                return string.Equals(eventSource.Name, rule.Name, StringComparison.Ordinal);
+            }
+            else
+            {
+                return IsEventSourceNameMatch(eventSource?.Name, rule?.Name);
+            }
+        }
+
+        /// <summary>
+        /// Returns true when eventSourceName satisfied the rule; false otherwise. Returns false when either is null.
+        /// </summary>
+        /// <param name="eventSourceName"></param>
+        /// <param name="rule"></param>
+        /// <returns></returns>
+        internal bool IsEventSourceNameMatch(string eventSourceName, string rule)
+        {
+            if (string.IsNullOrEmpty(eventSourceName) || string.IsNullOrEmpty(rule))
+            {
+                return false;
+            }
+
+            return eventSourceName.StartsWith(rule);
+        }
+
+        /// <summary>
+        /// When the event comes from a EventSource that matches the rule of DisabledSource, it should be dropped, unless the event source name matches one of the 
+        /// name exactly in the enabling rule.
+        /// </summary>
+        /// <param name="eventData">The event data to test.</param>
+        private bool IsDroppingEvent(EventWrittenEventArgs eventData)
+        {
+            string eventSourceName = eventData?.EventSource?.Name;
+            if (string.IsNullOrEmpty(eventSourceName))
+            {
+                throw new ArgumentNullException(nameof(eventSourceName));
```

Comment:
please do not throw

Reply:
I will change this to an event source event. Thanks!